File: backend/app/schemas/review.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field
from uuid import UUID

class ReviewBase(BaseModel):
    overall_rating: int = Field(..., ge=1, le=5)
    cleanliness_rating: Optional[int] = Field(None, ge=1, le=5)
    accessibility_rating: Optional[int] = Field(None, ge=1, le=5)
    accuracy_rating: Optional[int] = Field(None, ge=1, le=5)
    value_rating: Optional[int] = Field(None, ge=1, le=5)
    title: Optional[str] = Field(None, max_length=255)
    comment: Optional[str] = None

class ReviewCreate(ReviewBase):
    booking_id: UUID

class ReviewUpdate(BaseModel):
    overall_rating: Optional[int] = Field(None, ge=1, le=5)
    cleanliness_rating: Optional[int] = Field(None, ge=1, le=5)
    accessibility_rating: Optional[int] = Field(None, ge=1, le=5)
    accuracy_rating: Optional[int] = Field(None, ge=1, le=5)
    value_rating: Optional[int] = Field(None, ge=1, le=5)
    title: Optional[str] = None
    comment: Optional[str] = None

class ReviewResponse(ReviewBase):
    id: UUID
    booking_id: UUID
    parking_spot_id: UUID
    reviewer_id: UUID
    owner_response: Optional[str] = None
    owner_responded_at: Optional[str] = None
    helpful_count: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

class ReviewWithUserResponse(ReviewResponse):
    reviewer_name: str
    reviewer_image: Optional[str] = None

class ReviewOwnerResponse(BaseModel):
    response: str

class ReviewHelpful(BaseModel):
    review_id: UUID

class ReviewSummary(BaseModel):
    average_rating: float
    total_reviews: int
    rating_breakdown: dict  # {5: count, 4: count, ...}
    average_cleanliness: Optional[float] = None
    average_accessibility: Optional[float] = None
    average_accuracy: Optional[float] = None
    average_value: Optional[float] = None

File: mobile/src/types/review.ts
// Review types
export interface Review {
  id: string;
  booking_id: string;
  parking_spot_id: string;
  reviewer_id: string;
  overall_rating: number;
  cleanliness_rating?: number;
  accessibility_rating?: number;
  accuracy_rating?: number;
  value_rating?: number;
  title?: string;
  comment?: string;
  owner_response?: string;
  owner_responded_at?: string;
  helpful_count: number;
  created_at: string;
  updated_at: string;
}

export interface ReviewWithUser extends Review {
  reviewer_name: string;
  reviewer_image?: string;
}

export interface ReviewCreate {
  booking_id: string;
  overall_rating: number;
  cleanliness_rating?: number;
  accessibility_rating?: number;
  accuracy_rating?: number;
  value_rating?: number;
  title?: string;
  comment?: string;
}

export interface ReviewSummary {
  average_rating: number;
  total_reviews: number;
  rating_breakdown: Record<number, number>;
  average_cleanliness?: number;
  average_accessibility?: number;
  average_accuracy?: number;
  average_value?: number;
}
